src/allocator/memserver_allocator.cpp

```diff
@@ -107,6 +129,30 @@ void Memserver_Allocator::reset_profile() {
     MEMSERVER_PROFILE_START_TIME(NVMM)
 }
 void Memserver_Allocator::dump_profile() { NVMM_PROFILE_DUMP(); }
+
+void Memserver_Allocator::delayed_free_th(uint64_t thread_index) {
+    gc_th_struct_t *gc_th_obj = &delayed_free_thread_array[thread_index];
+    while (gc_th_obj->pthread_running == true) {
+        HeapInfo::iterator itr;
+        uint64_t nextId;
+        pthread_rwlock_rdlock(&gc_th_obj->rwLock);
+        while ((itr = gc_th_obj->heap_list->upper_bound(nextId)) !=
```

Comment:
The uninitialized first use of "nextId" @ line#139 might cause failures. "nextId" is assigned a value only at line#141.
I think the code block under the while loops can be re-arranged more effectively and simplify the use of pthread locks.